insights/parsers/logrotate_conf.py

```diff
@@ -0,0 +1,163 @@
+        True
+        >>> log_rt['/var/log/messages']['postrotate']
+        ['/sbin/killall -HUP syslogd']
+        >>> log_rt.options_of_logfile('/var/log/httpd/access.log')['mail']
+        'www@my.org'
+
+    Attributes:
+        data(dict): All parsed options and log files are stored in this
+            dictionary.
+    """
+
+    def parse_content(self, content):
+
+        def _parse_opts(line):
+            if '=' in line:
+                l_sp = line.split('=')
+            else:
+                l_sp = line.split(None, 1)
+            return l_sp[0], l_sp[1] if len(l_sp) == 2 else True
+
+        self.data = {}
+        self._options = []
+        self._log_files = []
+
+        log_opts = script = None
+        log_files = []
+        for line in get_active_lines(content):
+            if log_opts is None:
+                # not in log_file section
+                if line.lstrip(' \t\'"').startswith('/'):
+                    # log_file line
+                    log_files.extend([l.strip(' \t\'"') for l in line.rstrip('{').split()])
+                if line.endswith('{'):
+                    # start of the section
+                    log_opts = {}  # empty dict indicates necessary to parsing options
+                elif not log_files:
+                    # global options
+                    key1, val = _parse_opts(line)
+                    self.data[key1] = val
+                    self._options.append(key1)
+            else:
+                # in log_file section
+                if line.endswith('}'):
+                    # end of the section,
+                    # save options for each log_file individually
+                    for lf in log_files:
+                        self.data[lf] = log_opts
+                    # collect all log_files
+                    self._log_files.extend(log_files)
+                    log_files = []
+                    log_opts = None
+                    continue
+                if line in PAIRED_OPTS:
+                    # in script section
+                    script = line
+                    log_opts[script] = []
+                    continue
+                if line == 'endscript':
+                    # end of script section
+                    script = None
+                    continue
+                if script is not None:
+                    # scripts inner PARED_OPTS
+                    log_opts[script].append(line)
+                else:
+                    # common options in log_file section
+                    key, val = _parse_opts(line)
+                    log_opts[key] = val
+
+    @property
+    def options(self):
+        """
+        Return a list of global options configured in this configuration file.
+        """
+        return self._options
+
+    @property
+    def log_files(self):
+        """
+        Return a list of log files configured in this configuration file.
+        """
+        return self._log_files
```

Comment:
@xiangce would it make sense to just have `self.options` and `self.log_files` as attributes?

Reply:
The only reason I used the `@property` is trying to avoid writing data back to the object. But seems it's not a big deal at all. I will use attributes directly. Thanks @bfahr .